source/includes/_aviate-coupons.md

```diff
@@ -0,0 +1,299 @@
+
+---
+
+## Resources
+
+This section lists the models used by the Aviate Coupon APIs.
+
+### CouponInput
+
+Represents the input payload for creating a new coupon.
+
+| Name                 | Type       | Required | Description                                                                       |
+|----------------------|-----------|----------|-----------------------------------------------------------------------------------|
+| `redemptionCode`     | `string`  | yes      | Unique redemption code for the coupon                                            |
+| `reusable`           | `boolean` | yes      | Whether the coupon can be reused multiple times by the same customer             |
+| `discountType`       | `string`  | yes      | Discount type: `DISCOUNT_TYPE_FIXED` or `DISCOUNT_TYPE_PERCENTAGE`              |
+| `discountPrice`      | `string`  | no       | Fixed discount amount (required if `discountType` = FIXED)                      |
+| `discountCurrency`   | `string`  | no       | Currency for fixed discount (required if `discountType` = FIXED)                |
+| `discountPercentage` | `number`  | no       | Percentage discount (required if `discountType` = PERCENTAGE)                   |
+| `expirationDate`     | `string` (date-time) | no | Optional expiration date                                                         |
+| `planList`           | `array` of `string` | no | Optional list of plan names where the coupon is applicable                      |
+
+---
+
+### Coupon
+
+Represents a coupon that can be redeemed.
+
+
+| Name                 | Type       | Description                                                           |
+|----------------------|-----------|-----------------------------------------------------------------------|
+| `redemptionCode`     | `string`  | Unique redemption code                                               |
+| `reusable`           | `boolean` | Whether the coupon can be reused multiple times by the same customer|
+| `discountType`       | `string`  | Discount type                                                        |
+| `discountPrice`      | `string`  | Fixed discount amount                                                |
+| `discountCurrency`   | `string`  | Currency for fixed discount                                          |
+| `discountPercentage` | `number`  | Percentage discount                                                  |
+| `expirationDate`     | `string` (date-time) | Expiration date                                           |
+| `planList`           | `array`   | Applicable plan names                                                |
+| `createdDate`        | `string` (date-time) | Date when the coupon was created                               |
+| `archived`           | `boolean` | Whether the coupon is archived                                      |
+
```

Comment:
Is there a particular reason why we have `CouponInput` from `Coupon`? In all the other APIs, we do not have separate resource objects for the input hence asking.

Reply:
We have the [same thing for the Wallet](https://apidocs.killbill.io/aviate-wallet#walletinput), but I agree we may not be consistent across all our resources.